includes/class-freemius.php

```diff
@@ -14823,15 +14823,42 @@ function has_affiliate_program() {
             return $this->_plugin->has_affiliate_program();
         }
 
+        /**
+         * Get Plugin ID under which we will track affiliate application.
+         *
+         * This could either be the Bundle ID or the main plugin ID.
+         *
+         * @return number Bundle ID if developer has provided one, else the main plugin ID.
+         */
+        private function get_plugin_id_for_affiliate_terms() {
+            $bundle_id = $this->get_bundle_id();
+            if ( ! is_null( $bundle_id ) ) {
+                return $bundle_id;
+            }
+
+            return $this->_plugin->id;
+        }
+
         /**
          * @author Leo Fajardo (@leorw)
          * @since  1.2.4
          */
         private function fetch_affiliate_terms() {
             if ( ! is_object( $this->plugin_affiliate_terms ) ) {
-                $plugins_api     = $this->get_api_plugin_scope();
+                /**
+                 * In case we have a bundle set in SDK configuration, we would like to use that for affiliates, not the main plugin.
+                 */
+                if ( ! is_null( $this->get_bundle_id() ) ) {

```

Comment:
@swashata improvement suggestion to introduce a helper method named `has_bundle_context()` (or something along those lines) and use that instead of checking for `get_bundle_id()` every time.

Reply:
@swashata and I'd also recommend using `? :` instead of `if/else` here.

Reply:
@vovafeldman I didn't do the helper method because the abstraction seemed a little hasty to me.

- It is called only at 2 places at the moment.
- In one place we do need the `bundle_id` and in the other place, we don't. 
 
So it'd become

```php
private function has_bundle_context() {
  return ! is_null( $this->get_bundle_id() );
}

private function get_plugin_id_for_affiliate_terms() {
  return $this->has_bundle_context() ?
    $this->get_bundle_id() : // also called from `has_bundle_context`
    $this->_plugin->id;
}

private function fetch_affiliate_terms() {
  // ...
  $plugins_api = $this->has_bundle_context() ?
    $this->get_api_bundle_scope() :
    $this->get_api_plugin_scope();
  // ...
}
```

But NP, I will make the changes as I can see how this could improve readability.

Reply:
@vovafeldman Made the changes 👍